contrib/scripts/check-time.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: Apache-2.0
+# Copyright Authors of Cilium
+
+set -eu
+
+GOIMPORTS=("golang.org/x/tools/cmd/goimports" "-w")
+MATCH="^[^a-zA-Z]*\"time\"$"
+EXCLUDED_DIRS=(
+  # Wrapper directories
+  "time"
+
+  # Generated directories
+  "api"
+  "client"
+  "slim"
+
+  # Not Go source
+  ".git"
+  "_build"
+  "contrib"
+  "externalversions"
+  "examples"
+  "install"
+  "Documentation"
+
+  # Not for cilium-agent
+  "bugtool"
+  "cilium-dbg"
+  "clustermesh"
+  "clustermesh-apiserver"
+  "health"
+  "operator"
+  "plugins"
+  "tools"
+  "test"
+  "testutils"
+  "vendor"
+
+  # Source shared with other binaries
+  "hive"
+  "lock"
+
+  # Skip override (not applicable)
+  "defaults"
+  "loadinfo"
+  "logging"
+  "metric"
+  "probes"
+  "rand"
+  "types"
+
+  # May need subsequent evaluation to detect resiliency issues in CI
```

Comment:
Is this a todo?

Reply:
More or less yeah. The PR provides a good basis for now, there's some packages I stopped short of since they may have more nuanced interactions. I figured that for those, it'd be better to iterate on top once the initial bigger PR is agreed upon.

Added to the tasklist in https://github.com/cilium/cilium/issues/28844 .